macros/src/delegates.rs

```diff
@@ -12,7 +12,7 @@ type AsyncData = BoxFuture<Value, Error>;
 
 struct DelegateMethod<T, F> {
 	delegate: Arc<T>,
-	closure: F,
+	closure: Arc<F>,
```

Comment:
why not hold `T` and `F` by value but implement `RpcMethod` for `Arc<DelegateMethod<T, F>>`? Then only one clone has to be done per call